### PR TITLE
Vip6 feature identifier

### PR DESCRIPTION
**Pull request by afsmythe**
Addition of a `FeatureIdentifier` type to `ExternalGeospatialFeature`. 

This will move the feature index (previously `ShapeIdentifier`) to a `FeatureIdentifier` complex type, housing an `Index` to store the index value of the geospatial feature.

### CODE REVIEW

**Comment by afsmythe** [comment]
`FeatureIdentifier` is a repeating field so that multiple shapes can be assigned to a `Precinct`. Should `FeatureIdentifier` be encapsulated within a `FeatureIdentifiers` element? Or is it suitable that repeating `FeatureIdentifier` elements be nested directly underneath `ExternalGeospatialFeature`? E.g.

```
          <ExternalGeospatialFeature>
            <ExternalFileId>ef1</ExternalFileId>
            <FileFormat>shp</FileFormat>
            <FeatureIndentifier>
                <Index>3</Index>
            </FeatureIdentifier>
            <FeatureIdentifier>
                 <Index>4</Index>
            </FeatureIdentifier>
          </ExternalGeospatialFeature>
```
or
```
          <ExternalGeospatialFeature>
            <ExternalFileId>ef1</ExternalFileId>
            <FileFormat>shp</FileFormat>
            <FeatureIndentifiers>
                <FeatureIndentifier>
                  <Index>3</Index>
                </FeatureIdentifier>
                <FeatureIdentifier>
                   <Index>4</Index>
                </FeatureIdentifier>
            </FeatureIdentifiers>
          </ExternalGeospatialFeature>
```

File: docs/built_rst/tables/elements/feature_identifier.rst
Comment (jswiesner) [inline docs/built_rst/tables/elements/feature_identifier.rst:6]: WDYT about making this optional? In the future I could imagine a FeatureIdentifier element containing various different ways to refer to a shape (e.g. attribute-based identifier). In that case, it would be reasonable to provide the index or an attribute, in which case both would need to be optional in the spec to allow that flexibility.

File: docs/built_rst/xml/elements/precinct.rst
Comment (jswiesner) [inline docs/built_rst/xml/elements/precinct.rst:107]: typo: FeatureIndentifier

File: docs/built_rst/xml/elements/precinct.rst
Comment (jswiesner) [inline docs/built_rst/xml/elements/precinct.rst:150]: Is this intentional to pluralize ExternalGeospatialFeature?

**Comment by jswiesner** [comment]
> Should `FeatureIdentifier` be encapsulated within a `FeatureIdentifiers` element? Or is it suitable that repeating `FeatureIdentifier` elements be nested directly underneath `ExternalGeospatialFeature`?

I'm open to other thoughts, but I personally prefer the approach of 1 or more FeatureIdentifier elements nested directly underneath ExternalGeospatialFeature.

**Review by jswiesner — CHANGES_REQUESTED** [review]

File: docs/built_rst/tables/elements/feature_identifier.rst
Comment (afsmythe) [reply docs/built_rst/tables/elements/feature_identifier.rst:6]: Sounds good to me.

File: docs/yaml/elements/feature_identifier.yaml
Comment (jswiesner) [inline docs/yaml/elements/feature_identifier.yaml:6]: WDYT about making the index field an integer? When this used to be called "ShapeIdentifier", I originally chose a string format to offer greater flexibility for other shapefile formats. But since we plan to extend the FeatureIdentifier approach using other additional fields within the FeatureIdentifier element in the future (e.g. add an attribute field), we'll get flexibility that way instead. So this seems reasonable to me to be an integer field.

**Review by jswiesner — CHANGES_REQUESTED** [review]

File: docs/built_rst/tables/elements/feature_identifier.rst
Comment (JDziurlaj) [reply docs/built_rst/tables/elements/feature_identifier.rst]: Also what if the dereferenced `ExternalFile` only represents a single polygon? Should `FeatureIdentifier` be required at all?

Comment (jswiesner) [reply docs/built_rst/tables/elements/feature_identifier.rst]: Yes, FeatureIdentifier should still be required in that case. The ExternalGeospatialFeature is a reference to a single polygon within an external file containing one or more polygons. The FeatureIdentifier is how we know which single polygon to grab from that file. 

It's worth nothing that even if we make FeatureIdentifier.Index optional, one or more resolvable values is still required in a FeatureIdentifier element. This is a constraint that will need to be enforced in code to check 1) there is a usable value in each FeatureIdentifier and 2) each successfully resolves to one or more polygons in an external file.

File: docs/yaml/elements/feature_identifier.yaml
Comment (afsmythe) [reply docs/yaml/elements/feature_identifier.yaml:6]: SGTM! Thanks. Update forthcoming.

File: docs/built_rst/csv/elements/precinct.rst
Comment (jswiesner) [inline docs/built_rst/csv/elements/precinct.rst:170]: Looks like the CSV docs still need to be updated, the index field is still showing as a string type.

**Review by jswiesner — APPROVED** [review]
This LGTM, but I think the CSV rst still needs to be re-generated. Approving assuming the CSV docs will be fixed before merging. Thank you for putting this together!

**Pull request merged**
